common/test/views/support/cleaner/AmpEmbedCleanerTest.scala

```diff
@@ -16,6 +16,61 @@ class AmpEmbedCleanerTest extends FlatSpec with Matchers {
     document
   }
 
+  /////////////////////////////
+  // External video cleaner
+  /////////////////////////////
+
+  "AmpEmbedCleaner" should "replace an iframe in a YouTube video-element with an amp-youtube element" in {
+    val youtubeUrl = "https://www.youtube.com/watch?v=foo_12-34"
+    val doc = s"""<html><body><figure class="element-video" data-canonical-url="$youtubeUrl"><iframe></iframe></figure></body></html>"""
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+
+    result.getElementsByTag("amp-youtube").size should be(1)
+  }
+
+  "AmpEmbedCleaner" should "not create an amp-youtube element if videoid missing" in {
+    val youtubeUrl = s"http://www.youtube.com/"
```

Comment:
Looks like we have `http` and `https` for youtube in these tests, which makes me think: maybe we should only be accepting `https` video links?